lib/src/observable.dart

```diff
@@ -1,92 +1,131 @@
 import 'dart:async';
 import 'dart:ffi';
+import 'dart:isolate';
 
 import 'bindings/bindings.dart';
+import 'bindings/helpers.dart';
+import 'modelinfo/entity_definition.dart';
 import 'query/query.dart';
 import 'store.dart';
-import 'util.dart';
-
-// ignore_for_file: non_constant_identifier_names
-
-// dart callback signature
-typedef Any = void Function(Pointer<Void>, Pointer<Uint32>, int);
-
-class _Observable {
-  static final _anyObserver = <int, Pointer<OBX_observer>>{};
-  static final _any = <int, Map<int, Any>>{};
-
-  // sync:true -> ObjectBoxException: 10001 TX is not active anymore: #101
-  static final controller = StreamController<int>.broadcast();
-
-  // The user_data is used to pass the store ptr address
-  // in case there is no consensus on the entity id between stores
-  static void _anyCallback(
-      Pointer<Void> user_data, Pointer<Uint32> mutated_ids, int mutated_count) {
-    final storeAddress = user_data.address;
-    // call schema's callback
-    final storeCallbacks = _any[storeAddress];
-    if (storeCallbacks != null) {
-      for (var i = 0; i < mutated_count; i++) {
-        storeCallbacks[mutated_ids[i]]
-            ?.call(user_data, mutated_ids, mutated_count);
-      }
-    }
+
+/// Simple wrapper used below in ObservableStore to reduce code duplication.
+/// Contains shared code for single-entity observer and the generic/global one.
+class _Observer<StreamValueType> {
+  StreamController<StreamValueType> /*?*/ controller;
+  Pointer<OBX_observer> /*?*/ _cObserver;
+  ReceivePort /*?*/ receivePort;
+
+  int get nativePort => receivePort /*!*/ .sendPort.nativePort;
+
+  set cObserver(Pointer<OBX_observer> value) {
+    _cObserver = checkObxPtr(value, 'observer initialization failed');
+    _debugLog('started');
   }
 
-  static void subscribe(Store store) {
-    syncOrObserversExclusive.mark(store);
+  Stream<StreamValueType> get stream => controller /*!*/ .stream;
 
-    final callback = Pointer.fromFunction<obx_observer>(_anyCallback);
-    final storePtr = store.ptr;
-    _anyObserver[storePtr.address] =
-        bindings.obx_observe(storePtr, callback, storePtr.cast<Void>());
-    StoreCloseObserver.addListener(store, _anyObserver[storePtr.address], () {
-      unsubscribe(store);
-    });
+  _Observer() {
+    initializeDartAPI();
+  }
+
+  // start() is called whenever user starts listen()-ing to the stream
+  void init(void Function() start) {
+    controller = StreamController<StreamValueType>(
+        onListen: start, onPause: stop, onResume: start, onCancel: stop);
```

Comment:
`onListen` works for each new listener, and `onCancel` when all unsubscribed. is this taken into account? It would be nice if you wrote a test of how many times a method with multiple subscribers is called and whether all `_cObserver` are closed before being reinstalled.

Reply:
AFAIK you can only listen to a stream once (it's not a broadcast stream)

Reply:
Then there is another problem with accumulation `StreamController` for every `query` stream in memory

Reply:
> AFAIK you can only listen to a stream once (it's not a broadcast stream), I'll check though.

@vaind What if user do: `query.findStream().asBroadcastStream(onListen: (s){}, onCancel: (s)=> s.cancel())` in this case this methods will be called in the prime `StreamController` too as i remember. I think you should check it.

Reply:
Nope, the following code calls underlying stream onCancel() only once. 
```dart
final queryStream = query.findStream().asBroadcastStream(onListen: (s) { }, onCancel: (s) => s.cancel(); });
final subscription = queryStream.listen((list) {});
final subscription2 = queryStream.listen((list) {});
await subscription.cancel();
await subscription2.cancel();
```
Please create an issue (ideally with a reproducible example) if you find out observers/query streams don't work as expected.